src/main/java/uk/gov/justice/artemis/manager/connector/JmxArtemisConnector.java

```diff
@@ -49,6 +50,25 @@ public long remove(final String host, final String port, final String brokerName
         return removedMessages;
     }
 
+    @Override
+    public long reprocess(final String host, final String port, final String brokerName, final String destinationName, final Iterator<String> msgIds) throws Exception {
```

Comment:
Scanner and List both implement Iterator, so this is why Iterator was chosen in this instance.